src/main/java/messaging/MessageInteractor.java

```diff
@@ -0,0 +1,54 @@
+package messaging;
+
+import database.RoomDBGateway;
+import models.MessageModel;
+import models.RoomDBModel;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class MessageInteractor implements MessageInputBoundary {
+    private final MessageOutputBoundary presenter;
+    private final RoomDBGateway database;
+
+
+    public MessageInteractor(RoomDBGateway database, MessageOutputBoundary presenter) {
+        this.presenter = presenter;
+        this.database = database;
+    }
+
+    /**
+     * Handles the user request for sending a message.
+     */
+    @Override
+    public void handleSendMessage(MessageModel request) {
+        RoomDBModel room = database.getARoom(request.getRoomID());
+
+        // If message is empty
+        if (request.getContent() == null || request.getContent().isEmpty()) {
+            presenter.prepareMessageErrorView();
+        }
+
+        LocalDateTime timestamp = LocalDateTime.now();
+        // pass username to View like we do UserID vs. User Singleton vs. Fetch User to fix sender
+        String formattedMessage = "[" +
+                timestamp.format(DateTimeFormatter.ofPattern("HH:mm:ss")) + "] "
+                + request.getSender() + ": " + request.getContent() + "\n";
```

Comment:
Maybe here instead of `getSender();`  we fetch a username from the gateway based on `roomID`? We could also just store username in `View` like we store `userID`? Either way, we should be showing the `username` as sender not the `userID`.

Reply:
that's a good idea, we can circle back on that